gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -278,40 +319,286 @@ MRCompactorJobPropCreator getJobPropCreator(String topic, Path topicInputDir, Pa
     }
   }
 
-  private void processJob(State jobProps) {
-    this.futures.add(this.executorService.submit(getMRCompactorJobRunner(jobProps)));
+  private void processCompactionJobs() {
+    for (String inputFolder : this.inputFolderPriorities.keySet()) {
+      if (shouldVerifyDataCompleteness()) {
+        setInputFolderState(inputFolder, NO);
+        verifyDataCompleteness(inputFolder);
+      } else {
+        setInputFolderState(inputFolder, YES);
+      }
+      this.foldersBeingProcessed.add(inputFolder);
+    }
+    this.submitCompactionJobsAndWaitForCompletion();
+  }
+
+  private void verifyDataCompleteness(String inputFolder) {
+    this.dataComplVerifExecutor.get().execute(getDataComplVerifier(inputFolder, getInputFolderPriority(inputFolder)));
+  }
+
+  private boolean shouldVerifyDataCompleteness() {
+    return this.state.getPropAsBoolean(ConfigurationKeys.COMPACTION_SHOULD_VERIFY_DATA_COMPLETENESS,
```

Comment:
Please don't check at the configuration properties map in methods. Check only in the constructor and initialize the necessary member variables. Otherwise, figuring out all configuration that is used by a class is a major PITA. That is especially true when the configs have default values.